File: src/adapters/notifier/stdoutNotifier.ts
import { Notifier } from "../../ports/notifier.js";

export class StdoutNotifier implements Notifier {
  async notify(markdown: string): Promise<{ ok: boolean; status: number; body: string }> {
    console.log(markdown);
    return { ok: true, status: 200, body: "ok" };
  }
}

File: internal/adapters/notifier/stdout_notifier.go
package notifier

import (
	"context"
	"fmt"

	"github.com/HiroVodka/trendog/internal/ports"
)

type StdoutNotifier struct{}

func (StdoutNotifier) Notify(_ context.Context, markdown string) (ports.NotifyResult, error) {
	fmt.Println(markdown)
	return ports.NotifyResult{OK: true, Status: 200, Body: "ok"}, nil
}
